src/main/java/net/minecraftforge/client/textures/TextureAtlasSpriteLoaderManager.java

```diff
@@ -13,35 +13,43 @@
 import org.jetbrains.annotations.Nullable;
 
 import java.util.HashMap;
+import java.util.Map;
 
 /**
  * Manager for {@link ITextureAtlasSpriteLoader} instances.
  * <p>
  * Provides a lookup.
  */
-public final class TextureAtlasSpriteLoaderManager
-{
-    private static ImmutableMap<ResourceLocation, ITextureAtlasSpriteLoader> LOADERS;
+public final class TextureAtlasSpriteLoaderManager {
+    private static Map<ResourceLocation, ITextureAtlasSpriteLoader> getLoaders() {
+        final class LazyInit {
+            private static final Map<ResourceLocation, ITextureAtlasSpriteLoader> INSTANCE;
+
+            static {
+                var loaders = new HashMap<ResourceLocation, ITextureAtlasSpriteLoader>();
+                var event = new RegisterTextureAtlasSpriteLoadersEvent(loaders);
+                ModLoader.get().postEventWrapContainerInModOrder(event);
+                INSTANCE = ImmutableMap.copyOf(loaders);
+            }
+
+            private LazyInit() {}
+        }
+
+        return LazyInit.INSTANCE;
+    }
 
     /**
      * Finds the loader with the given name, or null if none is registered.
      */
     @Nullable
-    public static ITextureAtlasSpriteLoader get(ResourceLocation name)
-    {
-        return LOADERS.get(name);
+    public static ITextureAtlasSpriteLoader get(ResourceLocation name) {
+        return getLoaders().get(name);
     }
```

Comment:
So, I am not particularly a fan of this change. Because it is a behavior change. More specifically it means that modders can force the event to be fired by calling the get function. Instead of the modder getting a NPE {which we could make a more friendly "you call this too early" message) they get a potentially partially full map that can't be updated.

This one needs more thought as its not just a generic cleanup.